siliconcompiler/core.py

```diff
@@ -4987,6 +5037,130 @@ def __getstate__(self):
+
+        # Tool/task files
+        for keypath in self.allkeys('tool', tool, 'task', task):
+            if keypath[0] in ('output', 'input', 'report'):
+                continue
+            if 'default' in keypath:
+                continue
+            keypath = ['tool', tool, 'task', task, *keypath]
+            key_type = self.get(*keypath, field='type')
+            if 'file' in key_type or 'dir' in key_type:
+                self.set(*keypath, True, field='copy')
+
+        # Copy library files
+        for keypath in self.allkeys('library'):
+            if 'default' in keypath:
+                continue
+            keypath = ['library', *keypath]
+            key_type = self.get(*keypath, field='type')
+            if 'file' in key_type or 'dir' in key_type:
+                self.set(*keypath, include_libraries, field='copy')
+
+        # Copy pdk files
+        for keypath in self.allkeys('pdk'):
+            if 'default' in keypath:
+                continue
+            keypath = ['pdk', *keypath]
+            key_type = self.get(*keypath, field='type')
+            if 'file' in key_type or 'dir' in key_type:
+                self.set(*keypath, include_pdks, field='copy')
+
+        collect_path = os.path.join(issue_dir.name, 'collect')
+        self._collect(directory=collect_path)
+        additional_files[collect_path] = os.path.join(self.get('option','builddir'),
+                                                        self.get('design'),
+                                                        self.get('option', 'jobname'),
+                                                        'sc_collected_files')
+
+        git_data = {}
+        try:
+            # Check git information
+            repo = git.Repo(path=os.path.join(self.scroot, '..'))
+            git_data['commit'] = repo.head.commit.hexsha
+            git_data['date'] = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(repo.head.commit.committed_date))
+            git_data['author'] = f'{repo.head.commit.author.name} <{repo.head.commit.author.email}>'
+            git_data['msg'] = repo.head.commit.message
+            # Count number of commits ahead of version
+            version_tag = repo.tag(f'v{self.scversion}')
+            count = 0
+            for c in repo.head.commit.iter_parents():
+                count += 1
+                if c == version_tag.commit:
+                    break
+            git_data['count'] = count
+        except git.InvalidGitRepositoryError:
+            pass
+
+        issue_information = {}
+        issue_information['environment'] = {key: value for key, value in os.environ.items()}
+        issue_information['python'] = {"path": sys.path,
+                                       "version": sys.version}
+        issue_information['date'] = datetime.datetime.fromtimestamp(time.time()).strftime('%Y-%m-%d %H:%M:%S')
+        issue_information['machine'] = Chip._get_machine_info()
```

Comment:
Do we expect this is something that users might be using with the remote runs (and if so, would they be running `sc-issue` locally on results that come back, or getting an issue tarball straight from the server?) If yes to both, it might be nice to include whether the run was remote in the issue summary, so you know at a glance if the machine info captured here actually reflects the machine that ran the step.

Reply:
Once everything is working, I think it's something that the server should handle on a failure. At the moment it's meant for local runs